go/adbc/driver/flightsql/flightsql_statement.go

```diff
@@ -68,9 +75,27 @@ func (s *statement) Close() (err error) {
 
 // SetOption sets a string option on this statement
 func (s *statement) SetOption(key string, val string) error {
-	return adbc.Error{
-		Msg:  "[FlightSQL Statement] SetOption not implemented",
-		Code: adbc.StatusNotImplemented,
+	switch {
+	case key == OptionStatementQueueSize:
+		if size, err := strconv.Atoi(val); err != nil {
+			return adbc.Error{
+				Msg:  fmt.Sprintf("Invalid value for statement option '%s': '%s' is not a positive integer", OptionStatementQueueSize, val),
+				Code: adbc.StatusInvalidArgument,
+			}
+		} else if size <= 0 {
+			return adbc.Error{
+				Msg:  fmt.Sprintf("Invalid value for statement option '%s': '%s' is not a positive integer", OptionStatementQueueSize, val),
+				Code: adbc.StatusInvalidArgument,
+			}
+		} else {
+			s.queueSize = size
+		}
```

Comment:
the else is unnecessary since we return in the other cases. so let's simplify this and remove the else itself.